packages/vite/src/node/optimizer/index.ts

```diff
@@ -970,6 +962,21 @@ export function getDepsCacheDir(config: ResolvedConfig, ssr: boolean): string {
   return getDepsCacheDirPrefix(config) + getDepsCacheSuffix(config, ssr)
 }
 
+function getProcessingDepsCacheDir(config: ResolvedConfig, ssr: boolean) {
+  return (
+    getDepsCacheDirPrefix(config) +
+    getDepsCacheSuffix(config, ssr) +
+    getTempSuffix()
+  )
+}
+
+function getTempSuffix() {
+  return (
+    '_temp_' +
+    getHash(Date.now().toString() + Math.random().toString(16).slice(2))
```

Comment:
maybe process.pid instead of random works if we are sure that one process can only have one call of this at the same ms? Adding pid would prevent collisions in the unlikely event two different processes call it at the same ms and get the same random value (which may happen in a system with low entropy available, not sure about the way node implements random)